### PR TITLE
Accept extra_geom objects in notebook geometry API

### PR DESCRIPTION
**Pull request by takluyver**
geoAssembler wraps EXtra-geom's classes, but its wrappers aren't documented anywhere, and we've gradually moved functionality into EXtra-geom. So we should encourage users to use EXtra-geom objects in preference to geoAssembler objects. This aims to enable that.

Also:

- Better error messages when you pass no geometry for a detector which doesn't support that (although #30 will mean all detectors do)
- Fix copying the colormap on older versions of matplotlib, which don't have the `cmap.color()` method.
- Delete the 'local data' demo notebook, and describe the option to use local data in the main demo notebook. Having two copies of this means the secondary one gets out of date.

In response to #31

### CODE REVIEW

File: geoAssembler/geometry.py [hunk 1]
```diff
@@ -194,15 +194,25 @@ def write_quad_pos(self, filename):
     def wrap_extra_geom(geom_obj):
         """Wrap an EXtra-geom geometry object in the appropriate class
         """
+        filename = None
         if isinstance(geom_obj, GeometryAssembler):
             return geom_obj
+
         elif isinstance(geom_obj, AGIPD_1MGeometry):
             return AGIPDGeometry(geom_obj)
```
Comment (dallanto) [inline geoAssembler/geometry.py:202]: Not a code improvement suggestion, but a more general comment: I'm aware the different treatment if AGIPD is mainly due to the fact that we (DA, SPB) typically don't use HDF5 files, but rather CrystFEL geom files - but could we think of cases were HDF5 is still useful? It seems MID are using HDF5 geom for PyFAI notebooks ...

File: geoAssembler/io_utils.py [hunk 2]
```diff
@@ -11,30 +10,33 @@ def read_geometry(detector, filename, quad_pos=None):
...
+        quad_pos = Defaults.fallback_quad_pos.get(detector)
 
     if detector == 'AGIPD':
-        from .geometry import AGIPDGeometry
-        if os.path.isfile(filename):
+        if filename:
             return AGIPDGeometry.from_crystfel_geom(filename)
         else:
             return AGIPDGeometry.from_quad_positions(quad_pos)
     elif detector == 'LPD':
```
Comment (daviddoji) [inline geoAssembler/io_utils.py:26]: being very nit-picker, DSSC is everywhere define before LPD. In any case, it does not affect the code ;)

**Comment by daviddoji** [comment]
LGTM

File: geoAssembler/io_utils.py (hunk 2 above)
Comment (takluyver) [reply geoAssembler/io_utils.py:26]: Thanks, that's attention to detail! :laughing:

File: geoAssembler/geometry.py (hunk 1 above)
Comment (takluyver) [reply geoAssembler/geometry.py:202]: Yeah, I'm trying to get away from the assumption that we always use `.geom` files for AGIPD and always HDF5 files for DSSC and LPD. I've done some work on that as part of #30, though it doesn't cover AGIPD <-> HDF5 yet.

Comment (dallanto) [reply geoAssembler/geometry.py:202]: While I'll be looking at #30 as well, the comment is more to have a written note for the backlog, thanks for the answer (should we/I add that to the issues?)

Comment (takluyver) [reply geoAssembler/geometry.py:202]: Yup, particularly if you're aware of a specific need (e.g. adjusting AGIPD quadrant positions together with an EuXFEL HDF5 geometry file), please do make an issue.

If you know which interface the users in question would prefer to use, please make a note of that, too. :slightly_smiling_face:

Comment (dallanto) [reply geoAssembler/geometry.py:202]: Yes, I can ask MID even today, I guess, at the weekly ICI meeting (I strongly assume Johannes & Co, resp. their users have an affinity for Jupyter). Accordingly I can formulate an issue.

Comment (takluyver) [reply geoAssembler/geometry.py:202]: Thanks!

**Pull request merged**